cmake/Modules/FindMKL.cmake

```diff
@@ -304,7 +304,7 @@ if (USE_MKL AND USE_IDEEP)
   
   if (MKLDNN_INCLUDE_DIR)
     list(APPEND IDEEP_INCLUDE_DIR ${MKLDNN_INCLUDE_DIR})
-    list(APPEND __ideep_looked_for ${MKLDNN_INCLUDE_DIR})
+    list(APPEND __ideep_looked_for MKLDNN_INCLUDE_DIR)
```

Comment:
Actually, why do we do this? What's the error msg you saw if you treated it as a variable?

Reply:
cmake log:

-- Could NOT find IDEEP (missing:  /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include)
CMake Error at cmake/Modules/FindMKL.cmake:360 (message):
  Did not find IDEEP files!

Reply:
This doesn't look right. You should be able to find `MKLDNN` headers there, right? Could you print out `__ideep_looked` to see what it's looking for?

Reply:
$ cmake -version
cmake version 3.5.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).

-------------------------------------------

**without this fix:
before __ideep_looked_for: /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include
-- Could NOT find IDEEP (missing:  /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include)
after __ideep_looked_for: /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include
CMake Error at cmake/Modules/FindMKL.cmake:362 (message):
  Did not find IDEEP files!

------------------------------------------

**with this fix:
before __ideep_looked_for: MKLDNN_INCLUDE_DIR
-- Found IDEEP: /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include
after __ideep_looked_for: MKLDNN_INCLUDE_DIR
-- Found IDEEP (include: /home/jgu21/workspace/src/intel_pytorch/third_party/ideep/include;/home/jgu21/workspace/src/intel_pytorch/third_party/ideep/mkl-dnn/include, lib: /home/jgu21/workspace/src/intel_pytorch/zz/lib/libmkldnn.so)

Reply:
the find_package_handle_standard_args shoulld need var list, other than string lists...

Reply:
BTW, with presnet FindMKL.cmake, the libmkldnn will be linked with gomp, not iomp5.
This should be not expected, and the performance is not garuanteed.

Reply:
> BTW, with presnet FindMKL.cmake, the libmkldnn will be linked with gomp, not iomp5.
This should be not expected, and the performance is not garuanteed.

This is intentional. User should be able to choose to use gomp or iomp, but absolutely not both of them. I have benched the performance of using gomp and iomp and they are similar to me. Note that C2 can be included as a library and used by other applications. Situation is that if they use gomp and unknowingly linked in iomp from C2, they can have a nasty coredump and it's very difficult for them to figure it out without context.